### PR TITLE
feat: check if entity ids are unique for `incident=true`

### LINKED ISSUE #94: feat: handle merge error for `incident=True`
Getting merge error in `_add_incident_outcome()` when setting `incident=[True]` for cases where the outcome is not incident (e.g. one patient can have several cancer diagnoses, unlike t2d). 

Image

Handle this (keep only first row for each ID). The error shows up very late in the process so would be nice to let the user know that this is happening earlier on. 

**Solution**:

Skriv en test.
 
Og så i OutcomeSpec:
- Hvis `incident=True`, check at alle IDer i outcome (`dw_ek_borger`) er unikke. Hvis ikke, så raise error "Warning. You have set `incident=True`, but keys in dw_ek_borger are not unique. To handle this, timeseriesflattener will keep only the first 'incident' for each ID"

Og så i `timeseriesflattener/flattened_dataset.py` (måske omkring merge i `_add_incident_outcome()`):
- group `outcome_spec.values_df` by `dw_ek_borger`. Keep only first row for each ID

### PR DESCRIPTION
**Pull request by frillecode**
- [ ] I have assigned ranges (e.g. `>=0.1, <0.2`) to all new dependencies (allows dependabot to keep dependency ranges wide for better compatability)

Fixes #94.

Added a test for this now. Unsure about a sensible name for the test - up for suggestions!

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
Hey @frillecode, what is the status on this? :-)

The repo is in review, so very important to solve if we have an open bug.

File: src/timeseriesflattener/feature_spec_objects.py [hunk 1]
```diff
@@ -714,6 +716,16 @@ def is_dichotomous(self) -> bool:
 
         return len(self.values_df[col_name].unique()) <= 2  # type: ignore
 
+    def is_incident(self):
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_spec_objects.py:719]:
```suggestion
    def check_ids_are_unique_if_incident(self):
```

**Review by MartinBernstorff — APPROVED** [review]
Smukt Frida! Kun en lille ting jeg vil omdøbe, ellers ser det super ud! 💪

File: tests/test_timeseriesflattener/test_feature_spec_objects.py [hunk 2]
```diff
@@ -211,3 +217,29 @@ def test_predictorgroupspec_combinations_loader_kwargs():
...
+        resolve_multiple_fn="max",
+        incident=True,
+        fallback=np.NaN,
+        feature_name="value",
+    )
+
+    assert (
+        "Incident outcomes must have unique IDs. You have set incident=[True], however, the keys in your ID column are not unique. Consider setting incident=[False]."
+        in caplog.text
+    )
```
Comment (MartinBernstorff) [inline tests/test_timeseriesflattener/test_feature_spec_objects.py:245]: Her kan du med fordel bruge en context-manager ala. `with pytest.raises(ValueError, match=r"must have unique")`

File: src/timeseriesflattener/feature_spec_objects.py [hunk 3]
```diff
@@ -714,6 +716,16 @@ def is_dichotomous(self) -> bool:
...
+        """Check if the outcome is incident, i.e. contains unique IDs."""
+        if self.incident:
+            if self.values_df[self.entity_id_col_name].nunique() != len(
+                self.values_df,
+            ):
+                log.warning(
+                    "Incident outcomes must have unique IDs. You have set incident=[True], however, the keys in your ID column are not unique. Consider setting incident=[False].",
+                )
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_spec_objects.py:727]: Tænker det vil være godt reelt at raise en `ValueError` 👍

**Review by MartinBernstorff — CHANGES_REQUESTED** [review]

File: src/timeseriesflattener/feature_spec_objects.py (hunk 3 above)
Comment (frillecode) [reply src/timeseriesflattener/feature_spec_objects.py:727]: I stedet eller også?

Comment (frillecode) [reply src/timeseriesflattener/feature_spec_objects.py:727]: I stedet eller også?

File: src/timeseriesflattener/feature_spec_objects.py (hunk 1 above)
Comment (frillecode) [reply src/timeseriesflattener/feature_spec_objects.py:719]: Enig!

File: tests/test_timeseriesflattener/test_feature_spec_objects.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (frillecode) [reply tests/test_timeseriesflattener/test_feature_spec_objects.py:245]: God idé!

File: src/timeseriesflattener/feature_spec_objects.py (hunk 3 above)
Comment (MartinBernstorff) [reply src/timeseriesflattener/feature_spec_objects.py:727]: I stedet for, men med samme besked 👍

Comment (frillecode) [reply src/timeseriesflattener/feature_spec_objects.py:727]: I stedet eller også?

**Comment by MartinBernstorff** [comment]
@frillecode Har tilladt mig at simplificere testen lidt - læg mærke til, at `with pytest.raises` kun behøver at wrappe selve kaldet til den funktion, der raises fejlen - og at vi ikke behøver kigge i loggen, men den i stedet fanger exception'en og matcher på den :-)

**Review by MartinBernstorff — APPROVED** [review]

**Comment by frillecode** [comment]
Okay, hvor smart! Godt arbejde!

**Comment by HLasse** [comment]
Should we get this merged? Seems like it's basically ready?

**Pull request closed without merging**